lightning/src/chain/mod.rs

```diff
@@ -371,19 +371,22 @@ impl<T: Listen> Listen for dyn core::ops::Deref<Target = T> {
 	}
 }
 
-impl<T: core::ops::Deref, U: core::ops::Deref> Listen for (T, U)
+impl<T: core::ops::Deref, U: core::ops::Deref, V: core::ops::Deref> Listen for (T, U, V)
```

Comment:
Hmm...  this would break anyone needing only two `Listen` / `Confirm`. To support three, you just need to use `(T, (U, V))`.

Reply:
I don't think this breaks `Confirm` users, correct me if I'm wrong. 

Running into an issue with this approach because `Deref` is not implemented for the nested tuple. Still looking but LMK if you have any ideas on that.

Reply:
That's a fair point, my reasoning was that this would be a better default (which seems to be the purpose of this implementation) since most users will be using all three.

Reply:
> I don't think this breaks `Confirm` users, correct me if I'm wrong.

Right, I figured it was implemented for both, but `lightning-transaction-sync` uses `Vec` / `dyn` instead.

> That's a fair point, my reasoning was that this would be a better default (which seems to be the purpose of this implementation) since most users will be using all three.

We should keep it flexible, IMO. Otherwise, it would require atypical users to write their own implementation.